src/cursor.coffee

```diff
@@ -311,8 +311,29 @@ class Cursor extends Model
     @setBufferPosition([@getBufferRow(), 0])
 
   # Public: Moves the cursor to the beginning of the first character in the
-  # line.
+  # buffer line.
   moveToFirstCharacterOfLine: ->
+    # screenRow = @getScreenRow()
+    bufferRow = @getBufferRow()
+    bufferLineStart = @editor.clipBufferPosition([bufferRow, 0], skipSoftWrapIndentation: true)
+    bufferLineEnd = [bufferRow, Infinity]
+    bufferLineBufferRange = @editor.screenRangeForBufferRange([bufferLineStart, bufferLineEnd])
+    #
+    firstCharacterColumn = null
+    @editor.scanInBufferRange /\S/, bufferLineBufferRange, ({range, stop}) ->
+      firstCharacterColumn = range.start.column
+      stop()
+
+    if firstCharacterColumn? and firstCharacterColumn isnt @getBufferColumn()
+      targetBufferColumn = firstCharacterColumn
+    else
+      targetBufferColumn = bufferLineBufferRange.start.column
```

Comment:
@nathansobo it probably is. I modified that function from the original, which is now `moveToFirstCharacterOfScreenLine`. I'll try it locally and update it if it works.